add explainer for launch profiling (#9350)
getsentry/sentry-docs: docs/platforms/apple/common/profiling/index.mdx

The <PlatformIdentifier name="profiles-sample-rate" /> setting is _relative_ to the <PlatformIdentifier name="traces-sample-rate" /> setting.

</Note>

## Enable Launch Profiling

<Note>

This feature is experimental and may have bugs.

</Note>

_(New in version 8.21.0)_

> Normally, a profile can only be taken during a trace span after the SDK has been initialized. Now, when you call `SentrySDK.startWithOptions`, your configuration of sample rates for traces and profiles will be used to determine if the subsequent app launch should be automatically profiled. This allows you to gather information on what is going on in your app even before `main` is called, helping you issues you may have with slow app launches.

Author: Thanks, this looks great!

> If you use `SentryOptions.tracesSampler` or `SentryOptions.profilesSampler`, you can check `SentryTransactionContext.forNextAppLaunch` to decide whether or not to profile the next launch. Your customer sampler blocks will be invoked passing this context object at some point after calling `SentrySDK.startWithOptions`. If instead you simply set `SentryOptions.tracesSampleRate` and `SentryOptions.profilesSampleRate`, those numerical rates will be used.

Author: Hah, good catch on "customer sampler". But, it is something that happens automatically, and that they simply check. They don't make the invocation.

> If you use `SentryOptions.tracesSampler` or `SentryOptions.profilesSampler`, you can check `SentryTransactionContext.forNextAppLaunch` to decide whether or not to profile the next launch. Your customer sampler blocks will be invoked passing this context object at some point after calling `SentrySDK.startWithOptions`. If instead you simply set `SentryOptions.tracesSampleRate` and `SentryOptions.profilesSampleRate`, those numerical rates will be used.

Reviewer: Ah gotcha, how's this then?

```suggestion
If you use `SentryOptions.tracesSampler` or `SentryOptions.profilesSampler`, you can check `SentryTransactionContext.forNextAppLaunch` to see whether or not the next launch will be profiled. At some point after calling `SentrySDK.startWithOptions`, the SDK will invoke your custom sampler blocks by passing this context object.

If, instead, you set `SentryOptions.tracesSampleRate` and `SentryOptions.profilesSampleRate`, those numerical rates will be used to profile launches.
```

> If you use `SentryOptions.tracesSampler` or `SentryOptions.profilesSampler`, you can check `SentryTransactionContext.forNextAppLaunch` to decide whether or not to profile the next launch. Your customer sampler blocks will be invoked passing this context object at some point after calling `SentrySDK.startWithOptions`. If instead you simply set `SentryOptions.tracesSampleRate` and `SentryOptions.profilesSampleRate`, those numerical rates will be used.

Author: Still doesn't sound quite right, I took another crack at it:

```suggestion
If you use `SentryOptions.tracesSampler` or `SentryOptions.profilesSampler`, it will be invoked after you call `SentrySDK.startWithOptions`, with `SentryTransactionContext.forNextAppLaunch` set to `true` indicating that the result is for a launch profile sampling decision. If instead you simply set `SentryOptions.tracesSampleRate` and `SentryOptions.profilesSampleRate`, those numerical rates will be used directly.
```

> Currently, launch profiles are attached to a special performance transaction operation `app.launch` named simply `launch`.

Reviewer: ```suggestion
Currently, launch profiles are attached to a special performance transaction operation, `app.launch`, which you can refer to simply with `launch`.
```
"named" is kinda vague here. Not sure if it's how the user can reference it or if it's how the product displays it?

Author: It's how the product displays it, it's hardcoded in the Cocoa SDK.

Reviewer: ```suggestion
Currently, launch profiles are attached to a special performance transaction operation called `app.launch` and displayed in the product simply as `launch`.
```

Author: Yes that looks better!
